user-console/source/org/pentaho/mantle/client/ui/tabs/MantleTabPanel.java

```diff
@@ -818,7 +823,11 @@ public void onResponseReceived( Request request, Response response ) {
           if ( response.getStatusCode() == Response.SC_OK ) {
             JSONObject jsonObject = (JSONObject) JSONParser.parseLenient( response.getText() );
             if ( jsonObject != null ) {
-              isAsyncModeEnabled = jsonObject.get( "supportAsync" ).isBoolean().booleanValue();
+              if ( path.equals( REPORT_VIEWER_ASYNC_CONFIG_PATH ) ) {
```

Comment:
Make the method return a boolean, and you can avoid this whole "if" statement. 

    return jsonObject.get( "supportAsync" ).isBoolean().booleanValue();

and then when you *call* the method, assign the return value to the correct value. Return false in the catch-block in case there are any errors.